test/mpi/coll/test_coll_algos.sh

```diff
@@ -0,0 +1,31 @@
+nl='
+'
+
+coll_algo_tests=""
+
+########## Add tests for bcast algorithms ############
+
+#disable device collectives for bcast to test MPIR algorithms
+testing_env="env=MPIR_CVAR_BCAST_DEVICE_COLLECTIVE=0 "
+
+#test nb algorithms
+testing_env+="env=MPIR_CVAR_BCAST_INTRA_ALGORITHM=nb "
+testing_env+="env=MPIR_CVAR_IBCAST_DEVICE_COLLECTIVE=0 "
+algo_names="tree_kary tree_knomial"
```

Comment:
Can we automatically add new algorithms based on the CVAR definitions? I fear missing the step of adding a test for new implementations.

Reply:
That'd be great. Do we have a way to easily iterate over possible CVAR values?

Reply:
Also, there are certain algorithms that can only be used with certain sizes, etc. I'm afraid that if we don't do some sort of "by hand" selection, we'll end up with bad combinations (or a bunch of code here to weed out the bad ones.

Reply:
I'm not personally that familiar with it, but `maint/extractcvars.in` is probably the place to look. Specific sizes are a tougher problem. I suppose we can manage those by hand for the time-being.

Reply:
I don't think there's any input validation from the CVAR interface since these are just strings. If there were enums, we might be able to put bounds on the values. As it is, the only place where the valid values are listed is in the description. I don't know how we'd be able to parse that for valid input values.

Reply:
Yeah, fair point. Would the names of the functions be any better? Not ready to give up yet :smile:.

Reply:
Function names would be horrible. (I'll let you write the parser for that 😄.) But you might be able to use filenames.

See - https://github.com/pmodels/mpich/tree/master/src/mpi/coll/bcast

I believe the naming scheme is `<collective>-<intra|inter|allcomm>-<alg_name>.c`. That'll still have the same problem of certain algorithms not working for certain situations. E.g. `bcast_intra_smp.c` in the example above only works for intranode communicators.

Reply:
Yeah, that's another option. I think I'll just create an issue to solve this later. It would probably make a nice project for a new hire.

Reply:
Yeah, for now I would go for manual adding because testing is really algorithm specific. More than one CVAR may be required to test certain paths/variants in an algorithm and this is going to be specific to each algorithm. We can refine the approach later.